packages/@uppy/remote-sources/src/index.js

```diff
@@ -0,0 +1,72 @@
+import { BasePlugin } from '@uppy/core'
+import Dashboard from '@uppy/dashboard'
+import Dropbox from '@uppy/dropbox'
+import GoogleDrive from '@uppy/google-drive'
+import Instagram from '@uppy/instagram'
+import Facebook from '@uppy/facebook'
+import OneDrive from '@uppy/onedrive'
+import Box from '@uppy/box'
+import Unsplash from '@uppy/unsplash'
+import Url from '@uppy/url'
+import Zoom from '@uppy/zoom'
+
+import packageJson from '../package.json'
+
+const availablePlugins = [
+  Box,
+  Dropbox,
+  Facebook,
+  GoogleDrive,
+  Instagram,
+  OneDrive,
+  Unsplash,
+  Url,
+  Zoom,
+]
+
+export default class RemoteSources extends BasePlugin {
+  static VERSION = packageJson.version
+
+  constructor (uppy, opts) {
+    super(uppy, opts)
+    this.id = this.opts.id || 'RemoteSources'
+    this.type = 'acquirer'
+
+    const defaultOptions = {
+      sources: [
+        'Box',
+        'Dropbox',
+        'Facebook',
+        'GoogleDrive',
+        'Instagram',
+        'OneDrive',
+        'Unsplash',
+        'Url',
+      ],
+      target: Dashboard,
+    }
+    this.opts = { ...defaultOptions, ...opts }
+  }
+
+  setOptions (newOpts) {
+    this.uninstall()
+    super.setOptions(newOpts)
+    this.install()
+  }
+
+  install () {
+    this.opts.sources.forEach((pluginName) => {
+      const optsForRemoteSourcePlugin = { ...this.opts }
+      delete optsForRemoteSourcePlugin.sources
+      const plugin = availablePlugins.filter(p => p.name === pluginName)[0]
+      this.uppy.use(plugin, optsForRemoteSourcePlugin)
+    })
+  }
+
+  uninstall () {
+    this.opts.sources.forEach((pluginName) => {
+      const plugin = this.uppy.getPlugin(pluginName)
+      this.uppy.removePlugin(plugin)
+    })
```

Comment:
Shouldn't be worried that `this.opts.sources` can be modified by the user and the `uninstall` would not work correctly? Another (better?) way of handling it would be to have a private `Set` on the instance that caches all the plugins that were installed.

```suggestion
    this.#installedPlugins.forEach(plugin => this.uppy.removePlugin(plugin))
    this.#installedPlugins.clear()
```

Reply:
Yeah, it sounds better, not sure if Set is required or just a private array. But I did think of this, if you modify options the correct way via `uppy.getPlugin('RemoteSources').setOptions({ sources: ['Instagram'] })`, the plugin will first uninstall everything, then update options and install back: https://github.com/transloadit/uppy/pull/3676/files#diff-3b90938b025125a702e11b69f66acdd0c6985a892f64f4a28b06f0dcdb24e873R51-R55

Reply:
I think we should not assume that users will always use the correct way.

```js
const sources = ['Instagram']
uppy.use(RemoteSources, { sources })

sources.pop() // The user mutates the array for some reason

uppy.getPlugin('RemoteSources').uninstall() // The Instagram plugin is still installed
```

I think `Set` is more suited for our use case, but I suppose an array would also work.